### DISCUSSION TITLE
NextJs 12.2.3 middleware (after update nextJs to 12.2.3 can't make request inside middleware)

### DISCUSSION THREAD

**Discussion by eposha** [prisma/prisma#14473] [Q&A]
NextJs 12.2.3 middleware (after update nextJs to 12.2.3 can't make request inside middleware)

Has anyone experienced this problem?

**Comment by austincrim** [comment]
Hey @eposha 👋,

If you could share more details (code snippets, exact error messages, project setup), I would love to help!

**Reply by eposha** [reply]
@austincrim Hi

<---- schema.prisma ---->


```
datasource db {
  provider = "postgresql"
  url      = env("NEXT_PUBLIC_DATABASE_URL")
}

generator client {
  provider   = "prisma-client-js"
  engineType = "dataproxy"
}
```



<---- import prisma from 'lib/prisma'; ---->

```
import { PrismaClient } from '@prisma/client/edge';

let prisma: PrismaClient;

if (process.env.NODE_ENV === 'production') {
  prisma = new PrismaClient();
} else {
  let globalWithPrisma = global as typeof globalThis & {
    prisma: PrismaClient;
  };
  if (!globalWithPrisma.prisma) {
    globalWithPrisma.prisma = new PrismaClient();
  }
  prisma = globalWithPrisma.prisma;
}

export default prisma;
```



<---- middleware ---->

```
import { NextResponse } from 'next/server';
import type { NextRequest } from 'next/server';

import prisma from 'lib/prisma';

// RegExp for public files
const PUBLIC_FILE = /\.(.*)$/;

export const middleware = async (req: NextRequest) => {
  const url = req.nextUrl.clone();
  const { pathname } = url;

  if (
    pathname.startsWith('/_next') ||
    pathname.startsWith('/api') ||
    pathname.startsWith('/static') ||
    PUBLIC_FILE.test(pathname) ||
    pathname.includes('.')
  ) {
    const response = NextResponse.next();

    return response;
  }

  try {
    const user = await prisma.user.findFirst({
      where: {
        id: 'cl614s9q000100cpredrwjaml',
      },
    });

    console.log('user', user);
  } catch (error) {
    console.log(error);
  }
};

```

<---- error ---->
```

error - unhandledRejection: Error: aborted
    at connResetException (node:internal/errors:691:14)
    at abortIncoming (node:_http_server:596:17)
    at socketOnClose (node:_http_server:590:3)
    at Socket.emit (node:events:402:35)
    at TCP.<anonymous> (node:net:687:12) {
  code: 'ECONNRESET'
}
{ [Error: 
Invalid `prisma.user.findFirst()` invocation:


  Cannot fetch data from service:
fetch failed] code: 'P5010', clientVersion: '4.1.0', meta: undefined }
```


Next.js version 12.1.6 works fine

**Comment by austincrim** [comment]
It looks like we haven't changed anything here recently. I would recommend opening an issue on the [Next.js repo](https://github.com/vercel/next.js).

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
